src/layout/mod.rs

```diff
@@ -3745,6 +3792,60 @@ impl<W: LayoutElement> Layout<W> {
         monitor.move_workspace_up();
     }
 
+    pub fn set_workspace_name(&mut self, name: String) {
+        // also ignore the request if the name is already used by another workspace
+        if self.find_workspace_by_name(&name).is_some() {
+            return;
+        }
+
+        let Some(ws) = self.active_workspace_mut() else {
+            return;
+        };
+
+        ws.name.replace(name);
+
+        let wsid = ws.id();
+
+        // ensure last workspace is "unnamed"
+        match &mut self.monitor_set {
+            MonitorSet::Normal {
+                monitors,
+                active_monitor_idx,
+                ..
+            } => {
+                let monitor = &mut monitors[*active_monitor_idx];
+                if monitor
+                    .workspaces
+                    .last()
+                    .is_some_and(|last| last.id() == wsid)
+                {
+                    monitor.add_workspace_bottom();
+                }
+            }
+            MonitorSet::NoOutputs { ref mut workspaces } => {
+                if workspaces.last().is_some_and(|last| last.id() == wsid) {
+                    workspaces.push(Workspace::new_no_outputs(
+                        self.clock.clone(),
+                        self.options.clone(),
+                    ));
+                }
+            }
```

Comment:
`NoOutputs` workspaces must not have empty unnamed workspaces, so this is not needed.